core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -329,6 +332,13 @@ public boolean setReplication(Path path, short replication) throws IOException {
     }
   }
 
+  /**
+   * @param isShimFs set shim mode for this fs
+   */
+  public void setShimFs(boolean isShimFs) {
```

Comment:
it is tricky to have the setter for `isShimFs`, which may not prevent users to change this bool flag after a `FileSystem` instance created. If you see the usage of this method, it is always after `super()`. I would then just make `mIsShimFs` a `final` member variable, and take this argument on constructor to prevent others to change it afterwards.

Reply:
57dd443ea7110da95b1d3be4add7597f40f7a1a5

Reply:
Changed this further by @madanadit's feedback. https://github.com/Alluxio/alluxio/pull/9665/commits/cea90cf29590e4818a08ead26e6d9a90e13e57d2